java/ql/src/semmle/code/java/dataflow/internal/TaintTrackingUtil.qll

```diff
@@ -387,6 +390,11 @@ private predicate taintPreservingQualifierToMethod(Method m) {
       stringlist.getTypeArgument(0) instanceof TypeString
     )
   )
+  or
+  m instanceof StringFormatMethod
```

Comment:
I'm a bit surprised to see `StringFormatMethod` here - do any of them transfer taint from the qualifier? Also, using `StringFormatMethod` directly is possibly a poor fit in the context of data flow, since that class is all about capturing the formatting operation, and the collection of methods have very different data flow behaviours.

Reply:
`String.formatted` does, as well as the ones defined on `Formatter` (since they return their qualifier). But you're right, it's probably best to define the ones that are dataflow releavnt separately rather than using `StringFormatMethod`.